packages/ocaml/ocaml.4.10.1/opam

```diff
@@ -6,7 +6,7 @@ and polls it to initialise specific variables like `ocaml:native-dynlink`"""
 maintainer: "platform@lists.ocaml.org"
 depends: [
   "ocaml-config"
-  "ocaml-base-compiler" {= "4.10.1"} |
+  "ocaml-base-compiler" {= "4.10.1" | = "4.10.1+macos-arm64"} |
```

Comment:
Could it be moved into `ocaml-variants`, to my knowledge the distinction between `ocaml-base-compiler` and `ocaml-variants` is that the first only gathers the official release. This is not an official release so I would feel better not having it in `ocaml-base-compiler`. Does this would also get rid of this hack.

Reply:
I had an odd problem with `opam init` not working with ocaml-variants for some reason. I'll dig into it more later today.

Reply:
This does need to be in base-compiler since `opam init` specifically tries to initialise ocaml-system or ocaml-base-compiler (and not ocaml-variants).  This is close enough to an official release due to the unusual situation of having a new tier1 architecture show up, so I'll wait on @Octachron to take a decision on where the branch should live and then unblock macOS/ARM users.

Reply:
That seems really surprising to me. To my knowledge there is nothing in opam code that favours `opam-base-compiler` over `ocaml-variants`. Are you sure you didn't forget to remove the `ocaml-beta` dependency?

Reply:
https://github.com/ocaml/opam/blob/2.0.7/src/client/opamInitDefaults.ml#L26

Reply:
oh I see. Sorry I missed that.